system/Database/SQLSRV/PreparedQuery.php

```diff
@@ -34,6 +35,18 @@ class PreparedQuery extends BasePreparedQuery
      */
     protected $result;
 
+    /**
+     * A reference to the db connection to use.
+     *
+     * @var Connection
+     */
+    protected $db;
+
+    public function __construct(Connection $db)
+    {
+        parent::__construct($db);
+    }
```

Comment:
Is this to ensure a certain parameter type?

Reply:
Yes.
I added `getAllErrorMessages()` as `@internal` in `system/Database/SQLSRV/Connection.php`,
and use it.